src/utils/queryFacade.utils.ts

```diff
@@ -0,0 +1,146 @@
+    })
+    .catch(function (error) {
+      if (error.response) {
+        // The request was made and the server responded with a status code
+        // that falls out of the range of 2xx
+        console.log(error.response.data);
+        console.log(error.response.status);
+        console.log(error.response.headers);
+      } else if (error.request) {
+        // The request was made but no response was received
+        // `error.request` is an instance of XMLHttpRequest in the browser and an instance of
+        // http.ClientRequest in node.js
+        console.log(error.request);
+      } else {
+        // Something happened in setting up the request that triggered an Error
+        console.log('Error', error.message);
+      }
+      console.log(error.config);
+      return error.response.status;
+    });
+}
+
+/****************** RESPONSE CLEANER INTERFACE **********************/
+interface QueryCleaner {
+  (rawResponse: any): Vulnerability[];
+}
+
+//Cleaner instances
+let nvdCleaner: QueryCleaner;
+let sonatypeCleaner: QueryCleaner;
+
+//Cleaner implementaions
+nvdCleaner = function (rawResponse): Vulnerability[] {
+  let vulns: Vulnerability[] = [];
+  let rawVulns: any = rawResponse.vulnerabilities; //get all vulnerabilities
+
+  //Create Vulnerability for each cve in response
+  rawVulns.forEach(function (cve: any) {
+    let v: Vulnerability = {
+      cveId: cve['cve'].id,
+      packgaeId: -1,
+      impact: -1,
+      likelihood: -1,
+      risk: -1,
+      cvss2: '',
+    };
+    //Get correct CVSS version
+    if (cve['cve']['metrics'].hasOwnProperty('cvssMetricV2')) {
+      //V2
+      v.cvss2 =
+        cve['cve']['metrics']['cvssMetricV2'][0]['cvssData']['vectorString'];
+    } else if (cve['cve']['metrics'].hasOwnProperty('cvssMetricV31')) {
+      //V3
+      //map cvss2 to cvss3
+    }
+    vulns.push(v);
+  });
+  return vulns;
+};
+
+sonatypeCleaner = function (rawResponse): Vulnerability[] {
+  let vulns: Vulnerability[] = [];
+
+  let rawVulns: any = rawResponse[0].vulnerabilities;
+
+  //Create Vulnerability for each cve in response
+  rawVulns.forEach(function (cve: any) {
+    let v: Vulnerability = {
+      cveId: cve.id,
+      packgaeId: -1,
+      impact: -1,
+      likelihood: -1,
+      risk: -1,
+      cvss2: '',
+    };
+
+    //Get correct CVSS version
+    if (cve.hasOwnProperty('cvssVector')) {
+      let cvssVector: string = cve.cvssVector;
+
+      if (cvssVector.startsWith('CVSS:2')) {
+        //V2
+        v.cvss2 = cvssVector;
+      } else if (cvssVector.startsWith('CVSS:3')) {
+        //V3
+        //TO-DO:map cvss2 to cvss3
+        v.cvss2 = cvssVector;
+      }
+    }
+    vulns.push(v);
+  });
+
+  return vulns;
+};
+
+/****************** QUERY FACADE API **********************/
+//Sends one query, returns list of Vulnerabilities
+async function sendQuery(query: Query) {
+  return getVulnerabilities(query).then((response) => {
+    //Handle error
+    if (!isNaN(response)) {
```

Comment:
random question for personal curiosity: why would the response be a number when theres an error?

Reply:
Actually this is sloppy lol
I wanted your suggestion here - when axios gets an error response (aka not 2XX) I'm returning the status code from getVulnerabilities()

Do you have ideas? On error, I basically want to:
1. Log response code, message into our error logs from getVulnerabilities
2. return a value that lets calling functions know there was an error response
3. Consider having a framework in place for handling different types of error responses - like how we discussed retries

for number 3 we should consider if retries will be handled within QueryFacade or a decision made by calling components.